src/services/ldap/index.js

```diff
@@ -1,250 +1,272 @@
-         * @param {Object} options search options
-         * @return {Promise[Object]} resolves with object matching the query,
-         * rejects with error otherwise
-         */
-		searchObject(config, searchString, options) {
-			return this.searchCollection(config, searchString, options)
-				.then((objects) => {
-					if (objects.length > 0) {
-						return Promise.resolve(objects[0]);
-					}
-					return Promise.reject(new Error(`Object "${searchString}" not found`));
-				});
-		}
-
-		/**
-         * Returns all schools on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @return {Promise[Array[Object]]} resolves with all school objects or
-         * rejects with error
-         */
-		getSchools(config) {
-			return getLDAPStrategy(app, config).getSchools();
-		}
-
-		/**
-         * Returns all users at a school on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @param {School} school the school object
-         * @return {Promise[Object]} resolves with all user objects or rejects
-         * with error
-         */
-		getUsers(config, school) {
-			return getLDAPStrategy(app, config).getUsers(school);
-		}
-
-		/**
-         * Returns all classes at a school on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @param {School} school the school object
-         * @return {Promise[Object]} resolves with all class objects or rejects
-         * with error
-         */
-		getClasses(config, school) {
-			return getLDAPStrategy(app, config).getClasses(school);
-		}
-
-		/**
-         * Returns all experts on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @return {Promise[Object]} resolves with all expert objects or rejects
-         * with error
-         */
-		getExperts(config) {
-			const { searchString, options } = getLDAPStrategy(app, config).getExpertsQuery();
-			return this.searchCollection(config, searchString, options);
-		}
-	}
-
-	app.use('/ldap', new LdapService());
-	const systemService = app.service('/ldap');
-	systemService.hooks(hooks);
-};
+/* eslint-disable no-underscore-dangle */
+const ldap = require('ldapjs');
+const errors = require('@feathersjs/errors');
+const mongoose = require('mongoose');
+const hooks = require('./hooks');
+
+const getLDAPStrategy = require('./strategies');
+
+module.exports = function LDAPService() {
+	const app = this;
+
+	/**
+     * A service to communicate with LDAP servers.
+     *
+     * This service bundles common methods. Provider-specific functionality is
+     * delegated to strategies implementing a common interface (see
+     * `./strategies/interface.js`).
+     */
+	class LdapService {
+		constructor() {
+			this.clients = {};
+		}
+
+		/**
+		 * @deprecated
+		 */
+		find(params) {
+			// only needed to register as a feathers service
+		}
+
+		get(id, params) {
+			return app.service('systems').find({ query: { _id: id }, paginate: false })

```

Comment:
Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5713943)